.gitignore

```diff
@@ -127,3 +127,6 @@ dmypy.json
 
 # Pyre type checker
 .pyre/
+
+#documents for personal reference
+doc
```

Comment:
It's possible we'll want to include documentation under a folder such as `doc`, so might be best to use a different file/folder name for local comments.